Automattic/woocommerce-payments
# Add tracking for returning from OAuth connection

Fixes #943.

#### Changes proposed in this Pull Request

This PR adds a track `wcadmin_wcpay_stripe_connected` on returning from OAuth connection attempt. It also records whether the connection is a reconnect via `is_existing_stripe_account` event prop.

Additionally, this PR changes the `recordEvent` utility function to be wrapped in `domReady`. This guarantees all calls to the utility function does not precede `wcTracks` initialization.  

#### Testing instructions

**Prepping test on an existing connected site**
- Read the comment here: p90Yrv-2bd-p2#comment-5619

**Testing track via KYC (full flow)**
- Open your browser's developer console and enter `localStorage.setItem( 'debug', 'wc-admin:tracks*' );` to enable debugging for tracks.
- Go to WooCommerce > Payments, and click on "Set up" or "Finish Setup" button on the Connect Account page.
- Go through the connection process until you return from KYC.
- When ToS prompt appear, click on "Accept"
- Look at your developer console, it should have the debug for `wcadmin_wcpay_stripe_connected` track (look at screenshot below for example).
- It should have the `is_existing_stripe_account` prop set to `false`.

**Testing track via reconnect**
- Make sure your site is already connected.
- Using PHPMyAdmin (localhost:8083), delete `jetpack_options` in `wp_options` (essentially disconnects Jetpack).
- Go to WooCommerce > Payments, and click on "Set up" or "Finish Setup" button on the Connect Account page.
- Go through the connection process until you return from KYC.
- Look at your developer console, it should have the debug for `wcadmin_wcpay_stripe_connected` track (look at screenshot below for example).
- It should have the `is_existing_stripe_account` prop set to `true`.

![image](https://user-images.githubusercontent.com/3747241/118454923-75a29c80-b72b-11eb-84b7-fc719a8ad255.png)

-------------------
